### PR TITLE
QCut sample-based circuit cutting changleog update

### PR DESCRIPTION
**Pull request by anthayes92**
**Context:**

Simplifies sample-based circuit cutting additions and updates in changelog in preparation for release.

### CODE REVIEW

File: pennylane/transforms/__init__.py [hunk 1]
```diff
@@ -32,8 +32,6 @@
     ~transforms.classical_jacobian
     ~batch_params
     ~batch_input
-    ~draw
-    ~draw_mpl
```
Comment (anthayes92) [inline pennylane/transforms/__init__.py:36]: I'm not sure why I have had to remove these from recent PRs to get the docs to build both locally and for CI?

@christianbmadsen perhaps you have some insight on this? @josh146 mentioned these functions have been moved recently

Comment (anthayes92) [reply pennylane/transforms/__init__.py:36]: looks like my attempt to remove them this time is causing tests to fail :thinking:

Comment (albi3ro) [reply pennylane/transforms/__init__.py:36]: They were moved to the `drawer` module. I thought use of module code and conceptual similarity outweighed any structural similarities.  These functions were highly dependent on the `drawer` module, but nothing in the transforms module.  Nothing in the transforms module depended on them.

Comment (josh146) [reply pennylane/transforms/__init__.py:36]: it sounds like a merge conflict happened at some stage then, either locally @anthayes92 or in master

Comment (anthayes92) [reply pennylane/transforms/__init__.py:36]: Okay, so it sounds like they can be removed form here? 

Though it now looks like removing this is causing CI [tests to fail](https://github.com/PennyLaneAI/pennylane/runs/5859066500?check_suite_focus=true)

**Review by trbromley — COMMENTED** [review]
Thanks @anthayes92! Just left a few suggestions, and I also pushed some details about providing the classical processing fn.

File: doc/releases/changelog-dev.md [hunk 2]
```diff
@@ -104,47 +104,79 @@
...
+            qml.RX(1.3, wires=2)
+
+            qml.CNOT(wires=[0, 1])
+            qml.WireCut(wires=1)
+            qml.CNOT(wires=[1, 2])
+
+            qml.RX(x, wires=0)
+            qml.RY(0.7, wires=1)
+            qml.RX(2.3, wires=2)
+            return qml.sample(wires=[0, 2])
```
[the remark this answers is not in the document]
Comment (anthayes92) [reply doc/releases/changelog-dev.md]: fixed here: 43e1a977077cb206f94daa76e3c01573f158a964

**Review by trbromley — APPROVED** [review]

**Pull request merged**